app/lib/features/news/widgets/news_item.dart

```diff
@@ -144,4 +151,29 @@ class _NewsItemState extends ConsumerState<NewsItem> {
       ],
     );
   }
+
+  Widget newsActionButtons({
+    required NewsSlide newsSlide,
+  }) {
+    final newsReferencesList = newsSlide.references().toList();
+    if (newsReferencesList.isEmpty) return const SizedBox();
+
+    final referenceDetails = newsReferencesList.first.refDetails();
+    final uriId = referenceDetails.uri() ?? '';
+    final title = referenceDetails.title() ?? '';
+
+    if (title == NewsReferencesType.shareEvent.name) {
+      return ref.watch(calendarEventProvider(uriId)).when(
+            data: (calendarEvent) {
+              return EventItem(
+                event: calendarEvent,
+              );
+            },
+            loading: () => const EventItemSkeleton(),
+            error: (e, s) => Center(child: Text('Event failed: $e')),
+          );
+    } else {
+      return const SizedBox();
```

Comment:
Probably want to show some "unsupported - please upgrade" in this case for Future Action types, no?

Reply:
Ahh, good notice. Let me do needful here.